src/vcSettingsUI.cpp

```diff
@@ -762,80 +803,28 @@ void vcSettingsUI_VisualizationSettings(vcState *pProgramState, vcVisualizationS
     if (pVisualizationSettings->useCustomClassificationColours)
     {
       ImGui::SameLine();
-      if (ImGui::Button(udTempStr("%s##RestoreClassificationColors", vcString::Get("settingsRestoreDefaults"))))
-        memcpy(pVisualizationSettings->customClassificationColors, GeoverseClassificationColours, sizeof(pVisualizationSettings->customClassificationColors));
-
-      static const char *s_customClassifications[] =
-      {
-        "settingsVisClassUnclassified",
-        "settingsVisClassGround",
-        "settingsVisClassLowVegetation",
-        "settingsVisClassMediumVegetation",
-        "settingsVisClassHighVegetation",
-        "settingsVisClassBuilding",
-        "settingsVisClassLowPoint",
-        "settingsVisClassKeyPoint",
-        "settingsVisClassWater",
-        "settingsVisClassRail",
-        "settingsVisClassRoadSurface",
-        "settingsVisClassReserved",
-        "settingsVisClassWireGuard",
-        "settingsVisClassWireConductor",
-        "settingsVisClassTransmissionTower",
-        "settingsVisClassWireStructureConnector",
-        "settingsVisClassBridgeDeck",
-        "settingsVisClassHighNoise"
-      };
-
-      if (ImGui::Button(vcString::Get("settingsVisClassShowAll")))
-      {
-        for (size_t i = 0; i < udLengthOf(pVisualizationSettings->customClassificationToggles); i++)
-          pVisualizationSettings->customClassificationToggles[i] = true;
-      }
-
       ImGui::SameLine();
-      if (ImGui::Button(vcString::Get("settingsVisClassDisableAll")))
-      {
-        for (size_t i = 0; i < udLengthOf(pVisualizationSettings->customClassificationToggles); i++)
-          pVisualizationSettings->customClassificationToggles[i] = false;
-      }
+      if (ImGui::Button(udTempStr("%s##RestoreClassificationColors", vcString::Get("settingsRestoreDefaults"))))
+        memcpy(pProgramState->settings.visualization.customClassificationColors, GeoverseClassificationColours, sizeof(pProgramState->settings.visualization.customClassificationColors));
```

Comment:
I've seen a lot of `ud*` functions for memory management, but never for `memcpy`. Why is this?

Reply:
It's just not really necessary. Most other functions are wrapped because either they don't have a consistent API (and in some cases name, see `_udAlloc`) across the various platforms.